### ISSUE TITLE
Should this distribution offer generic type implementations?

### ISSUE THREAD

**Issue by johannessen** [johannessen/neo4j-types#7]
The text proposed in 1e8a758a38429078d40ef04741aeae0cdc7f3dd0 (#5) suggests that just about all implementation should eventually be removed from this distribution, leaving only the role itself.

However, the [Point](https://github.com/johannessen/neo4j-types/blob/main/lib/Neo4j/Types/Point.pod) type already exists as what is basically a self-contained type, thanks to the existence of the `new()` constructor. So, with 1e8a758a38429078d40ef04741aeae0cdc7f3dd0 removing the documentation of the data structure itself, we already have achieved encapsulation – as far as you can get with old-style Perl, anyway.

We _could_ decide to give Node, Relationship, and Path the same treatment: Offer a `new()` constructor for them, so that drivers can choose to use Neo4j::Types’ implementation after all without needing to know any internals.

Question is: Would they make that choice?

The generic type might well make sense for Point, which can be constructed by users and passed on to Neo4j as a parameter. (In theory, anyway – as of right now, no Perl module actually supports this.)

But for Node, Relationship, and Path I’m not sure I really see this use case. And even consistency is a weak argument because these types _are_ different from Point.

So I’d tend towards _maybe_ offering a generic implementation for Point, but not for the others.

**Comment by johannessen** [comment]
There’s nothing wrong with a generic **Node / Relationship / Path** implementation as such, but there is no reason the role itself should provide one. If such a generic implementation exists, better to have additional modules for that. In fact, those should probably be in another distribution entirely, so that users who don’t need them don’t even need to install them.

However, for **Point** and other Cypher property types, there has to be a way to create them locally. Given that, offering a generic implementation would actually be useful: It might free drivers authors from having to implement and document a `new()` constructor; users could just use ["new" in Neo4j::Types::Point](https://metacpan.org/pod/Neo4j::Types::Point#new). Additionally, it would give driver authors an incentive to accept Neo4j::Types::Point values created *elsewhere*; e.g. ["run_query" in Neo4j::Bolt::Cxn](https://metacpan.org/pod/Neo4j::Bolt::Cxn#run_query($cypher_query,-[$param_hash])) should eventually accept [Neo4j::Driver::Type::Point](https://metacpan.org/pod/Neo4j::Driver::Type::Point) values.

Bottom line: No change required here.
